platform/binance/block.go

```diff
@@ -2,38 +2,53 @@ package binance
 
 import (
 	"github.com/trustwallet/blockatlas/pkg/blockatlas"
-	"github.com/trustwallet/blockatlas/pkg/errors"
 )
 
 func (p *Platform) CurrentBlockNumber() (int64, error) {
-	// No native function to get height in explorer API
-	// Workaround: Request list of blocks
-	// and return number of the newest one
-	list, err := p.client.GetBlockList(1)
+	info, err := p.client.fetchNodeInfo()
 	if err != nil {
 		return 0, err
 	}
-	if len(list.BlockArray) == 0 {
-		return 0, errors.E("no block descriptor found", errors.TypePlatformApi)
-	}
-	return list.BlockArray[0].BlockHeight, nil
+
+	return info.SyncInfo.LatestBlockHeight, nil
 }
 
 func (p *Platform) GetBlockByNumber(num int64) (*blockatlas.Block, error) {
-	srcTxs, err := p.client.GetBlockByNumber(num)
+	blockTxs, err := p.client.GetBlockTransactions(num)
 	if err != nil {
 		return nil, err
 	}
 
-	var txs blockatlas.TxPage
-	childTxs, err := p.getTxChildChan(srcTxs.Txs)
-	if err == nil {
-		txs = NormalizeTxs(childTxs, "", "")
-	} else {
-		txs = NormalizeTxs(srcTxs.Txs, "", "")
+	childTxs := make([]Tx, 0)
+	for _, t := range blockTxs {
+		if len(t.SubTransactions) > 0 {
+			for _, tSub := range t.SubTransactions {
+				childTxs = append(childTxs, normalizeBlockSubTx(&t, &tSub))
+			}
+		}
+	}
+
+	txs := NormalizeTxs(childTxs, "")
+
+	return &blockatlas.Block{Number: num, Txs: txs}, nil
+}
+
+func normalizeBlockSubTx(t *TxV2, tSub *SubTx) Tx {
```

Comment:
Maybe normalizeBlockSubTx(t TxV2, tSub SubTx)?
Because you don't change t and tSub in this func

Reply:
Bit modified this function, please take a look.